src/main/kotlin/exercise4/task2/TransactionalBankAccount.kt

```diff
@@ -78,8 +79,104 @@ import java.util.*
  *      ```
  */
 
+enum class TType {
```

Comment:
nit: Please don't use abbreviations in public types. Naming enum `TransactionType` instead of `TType` give clear understanding to the reader on what does enum type represents.